### PR TITLE
op-batcher: Simplify configs, dedup init, & simplify init code

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

This uses the same op-services/crypto signer initialization. It also simplifies the batcher struct and config.

File names are kept the same & structs aren't really moved around to reduce the diff.

**Metadata**
- Fixes ENG-3210

### CODE REVIEW

**Comment by tynes** [comment]
Looks like the build is failing, not able to import `"github.com/ethereum-optimism/optimism/op-proposer/txmgr"`

File: op-batcher/batcher/batch_submitter.go [hunk 1]
```diff
@@ -30,98 +27,75 @@ const (
...
-		var batchSubmitter *BatchSubmitter
-		if !cfg.SignerConfig.Enabled() {
-			bs, err := NewBatchSubmitter(cfg, l)
-			if err != nil {
-				l.Error("Unable to create Batch Submitter", "error", err)
-				return err
-			}
-			batchSubmitter = bs
-		} else {
-			signerClient, err := opsigner.NewSignerClientFromConfig(l, cfg.SignerConfig)
```
Comment (zhwrd) [inline op-batcher/batcher/batch_submitter.go:52]: seems like support for signer service deleted in this pr?

Comment (zhwrd) [reply op-batcher/batcher/batch_submitter.go:52]: ah nvm, see it got refactored to op-services/crypto

**Review by sebastianst — APPROVED** [review]
LGTM

**Review by protolambda — APPROVED** [review]

**Pull request merged**